executor/builder.go

```diff
@@ -2177,6 +2185,22 @@ func markChildrenUsedCols(outputSchema *expression.Schema, childSchema ...*expre
 	return
 }
 
+// extractChildrenUsedColIdxs extract column indexes of child executors from children used columns mark
+func extractChildrenUsedColIdxs(childrenUsedColsMark [][]bool) [][]int {
```

Comment:
Should the loop like [this](https://github.com/pingcap/tidb/blob/master/executor/joiner.go#L142-L147) be replaced by this function?

Reply:
Good idea! I am not so sure that replace could done in this PR?

Reply:
It's ok to do it in this PR